src/system_identity.rs

```diff
@@ -1 +1,235 @@
+pub mod creation_response;
+pub mod generator;
+pub mod iam_client;
+pub mod identity_creator;
 pub mod input_data;
+
+/// System identity information. Final output of the System Identity creation process.
+#[derive(Debug, Clone)]
+pub struct SystemIdentity {
+    pub id: String,
+    pub name: Option<String>,
+    pub client_id: String,
+    pub organization_id: String,
+    pub identity_type: SystemIdentityType,
+}
+
+#[derive(Debug, Clone, PartialEq)]
+pub enum SystemIdentityType {
+    L1 {
+        client_secret: ClientSecret,
+        credential_expiration: String,
+    },
+    L2 {
+        pub_key: Base64PublicKey,
+    },
+}
```

Comment:
maybe a rebase is needed since I see it already merged, however does it need some docs in code exmplaining L1 vs L2? or is it somewhere else already?